### PR TITLE
decoupling some tests scripts from manifest

### PR DESCRIPTION
**Pull request by changev**
https://rackhd.atlassian.net/browse/RAC-4706
Decoupling test scripts from manifest.
Make implementing none-manifest test more easier.

@panpan0000 @PengTian0

### CODE REVIEW

**Comment by panpan0000** [comment]
same, please provide what you have tested and how to prove your test is good.

File: jobs/function_test/function_test.groovy [hunk 1]
```diff
@@ -83,32 +105,36 @@ def function_test(String test_name, String label_name, String TEST_GROUP, Boolea
...
+                            junit 'RackHD/test/*.xml'
+                            int failure_count = 0
+                            int error_count = 0
+                            if(fileExists ("downstream_file")) {
+                                def props = readProperties file: "downstream_file"
+                                failure_count = "${props.failures}".toInteger()
+                                error_count = "${props.errors}".toInteger()
+                            }
+                            if (failure_count > 0 || error_count > 0){
+                                currentBuild.result = "SUCCESS"
```
Comment (panpan0000) [inline jobs/function_test/function_test.groovy:134]: why there's failure but result is SUCCESS

**Comment by PengTian0** [comment]
Why not just set ```SKIP_PREP_DEP=true``` for post test?

File: jobs/function_test/function_test.groovy (hunk 1 above)
Comment (changev) [reply jobs/function_test/function_test.groovy:134]: This is just a code copy. 
@PengTian0 to answer this question.

**Comment by changev** [comment]
@PengTian0 
Some steps in prepare.sh are still need in post test or other test, so can't just set SKIP_PREP_DEP=true.

Extract common prepare steps is better IMO.

**Comment by changev** [comment]
test this please

**Pull request closed without merging**